Source/Scene/TimeDynamicPointCloud.js

```diff
@@ -0,0 +1,554 @@
+define([
+        '../Core/Check',
+        '../Core/combine',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/getTimestamp',
+        '../Core/JulianDate',
+        '../Core/Math',
+        '../Core/Matrix4',
+        '../Core/Resource',
+        './ClippingPlaneCollection',
+        './PointCloud',
+        './PointCloudEyeDomeLighting',
+        './PointCloudShading',
+        './SceneMode',
+        './ShadowMode'
+    ], function(
+        Check,
+        combine,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        getTimestamp,
+        JulianDate,
+        CesiumMath,
+        Matrix4,
+        Resource,
+        ClippingPlaneCollection,
+        PointCloud,
+        PointCloudEyeDomeLighting,
+        PointCloudShading,
+        SceneMode,
+        ShadowMode) {
+    'use strict';
+
+    /**
+     * Provides functionality for playback of time-dynamic point cloud data.
+     *
+     * @alias TimeDynamicPointCloud
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {Clock} options.clock A {@link Clock} instance that is used when determining the value for the time dimension.
+     * @param {TimeIntervalCollection} options.intervals A {@link TimeIntervalCollection} with its data property being an object containing a uri to a Point Cloud (.pnts) tile and an optional transform.
+     * @param {Boolean} [options.show=true] Determines if the point cloud will be shown.
+     * @param {Matrix4} [options.modelMatrix=Matrix4.IDENTITY] A 4x4 transformation matrix that transforms the point cloud.
+     * @param {ShadowMode} [options.shadows=ShadowMode.ENABLED] Determines whether the point cloud casts or receives shadows from each light source.
+     * @param {Number} [options.maximumMemoryUsage=256] The maximum amount of memory in MB that can be used by the point cloud.
```

Comment:
Perhaps details for the property reference doc but how does this work if one frame exceeds the usage?  I assume it is allowed.  Reference doc should also briefly explain that this is streaming, prefectches, and uses an LRU cache like the GitHub comment mentions.

Reply:
One frame that exceeds the usage will still be rendered.

I added a note that should clarify this (similar to the `Cesium3DTileset` wording):
> Frames that are not being loaded or rendered are unloaded to enforce this.